erpnext/patches/v13_0/create_leave_policy_assignment_based_on_employee_current_leave_policy.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2019, Frappe and Contributors
+# License: GNU General Public License v3. See license.txt
+
+from __future__ import unicode_literals
+
+import frappe
+
+def execute():
+    employees_with_leave_policy = frappe.db.sql("SELECT name, leave_policy FROM `tabEmployee` WHERE leave_policy IS NOT NULL", as_dict = 1)
+
+    for employee in employees_with_leave_policy:
+        print(employee.name)
+        if not frappe.db.exists("Leave Allocation", {"employee":employee.name, "leave_policy": employee.leave_policy, "docstatus": 1}):
+            create_assignment(employee.name, employee.leave_policy)
+
+
+    employee_grade_with_leave_policy = frappe.db.sql("SELECT name, default_leave_policy FROM `tabEmployee Grade` WHERE default_leave_policy IS NOT NULL", as_dict = 1)
+
+    for grade in employee_grade_with_leave_policy:
+        employees = get_employee_with_grade(grade.name)
```

Comment:
If leave_policy mentioned in both Employee and Grade, in that case Employee record should get priority. In your case, it will create a duplicate assignment.

Reply:
done